sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/CommunicationTokenCredentialTests.java

```diff
@@ -309,6 +309,23 @@ public void refresherShouldBeCalledImmediatelyWithoutInitialToken() throws Inter
         tokenCredential.close();
     }
 
+    @Test
+    public void refresherShouldBeCalledOnDemandWithoutInitialToken() throws InterruptedException, ExecutionException, IOException {
+        immediateFresher.resetCallCount();
+        String tokenStr = tokenMocker.generateRawToken("resourceId", "userIdentity", 15 * 60);
+        immediateFresher.setRefreshedToken(tokenStr);
+        CommunicationTokenRefreshOptions tokenRefreshOptions = new CommunicationTokenRefreshOptions(immediateFresher)
+            .setRefreshProactively(false);
+        CommunicationTokenCredential tokenCredential = new CommunicationTokenCredential(tokenRefreshOptions);
+        assertEquals(0, immediateFresher.numCalls());
+        StepVerifier.create(tokenCredential.getToken())
+            .assertNext(token -> {
+                assertFalse(token.isExpired(), "New AccessToken should have been generated on demand");
```

Comment:
what about checking if the refreshed token is the same as tokenStr provided to the callback and the callback called only once?

Reply:
Updated